### PR TITLE
Added Float32 and Float64 consts

### LINKED ISSUE #4785: Floating point numbers constants
Currently, `Float32` and `Float64` data types both provide `NAN`, `INFINITY`, `MIN`, and `MAX` constants, with `MIN = -INFINITY` and `MAX = INFINITY`.

In order to provide meaningful values for floating-point arithmetics, I suggest: 
1. To change `MIN`and `MAX` respectively to the smallest finite value and the largest finite value that can be represented.
2. To add extra constant values, such as:
    - `EPSILON`: the machine epsilon (difference between `1.0` and the next representable value).
    - `DIG` : the number of decimal digits that can be represented without losing precision.
    - `RADIX`: the radix or integer base used by the internal representation.
    - `MANT_DIG`: the number of digits that can be represented without losing precision (in base `RADIX`).
    - `MIN_EXP`: the minimum possible normal power of 2 exponent.
    - `MAX_EXP`: the maximum possible normal power of 2 exponent.
    - `MIN_10_EXP`: the minimum possible power of 10 exponent (such that `10**MIN_10_EXP` is representable).
    - `MAX_10_EXP`: the maximum possible power of 10 exponent (such that `10**MAX_10_EXP` is representable).
    - `MIN_NORMAL` or `MIN_POSITIVE`: smallest representable normal value that is not 0.

### PR DESCRIPTION
**Pull request by konovod**
Fixes #4785
Adds `EPSILON`, `DIGITS`, `RADIX`, `MANT_DIG`, `MIN_EXP`, `MAX_EXP`, `MIN_10_EXP`, `MAX_10_EXP` and `MIN_POSITIVE` consts to Float32 and Float64.
I've renamed DIG -> DIGITS as it imho looks clearer this way. For a `ROUNDS` - I don't know how to get its value and is it really needed, so skipped it.
If the constants are different for some platforms, they of course should be moved to platform-specific files, but i think they are currently same for all supported platforms.

### CODE REVIEW

**Comment by konovod** [comment]
Huh, travis fails on 32 bits with strange error. No idea what could i broke (and no 32-bit machine to test) - `Float64::MAX and `Float64::MIN` are used only in `benchmark/ips.cr` and changing MAX/MIN to the finite value is the only potentially breaking change.

**Comment by ysbaddaden** [comment]
Try to not change MIN and MAX.

**Comment by konovod** [comment]
Still broken on 32-bits. No ideas.

**Comment by konovod** [comment]
Well, working now. I'll try to revert `MAX`/`MIN` back to finite values.

**Comment by konovod** [comment]
Reverted, CI is green :smile:

**Comment by RX14** [comment]
It would be great if you could specify the float constants as hexadecimal then use `unsafe_as` to convert them to floats. That way we know we're not loosing precision.

**Comment by konovod** [comment]
@RX14 in that case, should there also be some specs ensuring that `0x34000000_u32.unsafe_as(Float32)` is actually small number?

**Comment by RX14** [comment]
@konovod yes.

**Comment by konovod** [comment]
done, but there is an error on 32-bits travis again. If it happens on other PRs too it's ok (well, not ok, but unrelated), but if only on this - perhaps there is some problem with it?

**Comment by wontruefree** [comment]
@konovod what is the status of this PR?

**Comment by konovod** [comment]
I don't see anything preventing it from merge. Travis 32-bit was failing with apparently irrelevant error, perhaps will be green if restarted. But i forgot how to restart CI - should I force push the same commit?

File: src/float.cr [hunk 1]
```diff
@@ -102,8 +102,29 @@ end
...
-  MAX      = INFINITY.as Float32
+  # Smallest finite value
+  MIN = 0xff7fffff_u32.unsafe_as(Float32) # -3.40282347e+38_f32
+  # Largest finite value
+  MAX = 0x7f7fffff_u32.unsafe_as(Float32) # 3.40282347e+38_f32
+  # The machine epsilon (difference between 1.0 and the next representable value)
+  # value is
+  EPSILON = 0x34000000_u32.unsafe_as(Float32) # 1.19209290e-07_f32
+  # The number of decimal digits that can be represented without losing precision
+  DIGITS = 15
```
Comment (RX14) [inline src/float.cr]: This should be 6.

**Comment by wontruefree** [comment]
This might sounds like a dumb question but are these set at compile time or runtime?  seem to me things like `unsafe_as` would be runtime.  If this is the case should these be changes to compile time?  Or does it even matter?

File: src/float.cr (hunk 1 above)
Comment (konovod) [reply src/float.cr]: nice catch, my bad!

**Comment by konovod** [comment]
Even if `unsafe_as` isn't optimized away, consts initialization happens only at start of program, so performance doesn't matter. Anyway, I've changed it to the way @RX14 suggested, so `unsafe_as` is now only in specs.

File: src/float.cr [hunk 2]
```diff
@@ -107,8 +107,29 @@ end
...
-  MAX      = INFINITY.as Float32
+  # Smallest finite value
+  MIN = -3.40282347e+38_f32
+  # Largest finite value
+  MAX = 3.40282347e+38_f32
+  # The machine epsilon (difference between 1.0 and the next representable value)
+  # value is
+  EPSILON = 1.19209290e-07_f32
+  # The number of decimal digits that can be represented without losing precision
+  DIGITS = 15
```
Comment (Sija) [inline src/float.cr]: Still `6` :)

Comment (konovod) [reply src/float.cr]: huh

File: src/float.cr [hunk 3]
```diff
@@ -107,8 +107,29 @@ end
...
   NAN      = (0_f32 / 0_f32).as Float32
   INFINITY = (1_f32 / 0_f32).as Float32
-  MIN      = (-INFINITY).as Float32
-  MAX      = INFINITY.as Float32
+  # Smallest finite value
+  MIN = -3.40282347e+38_f32
+  # Largest finite value
+  MAX = 3.40282347e+38_f32
+  # The machine epsilon (difference between 1.0 and the next representable value)
+  # value is
```
Comment (luislavena) [inline src/float.cr]: Probably leftover from a change?

**Comment by bcardiff** [comment]
Thanks @konovod !

**Pull request merged**